piquasso/api/simulator.py

```diff
@@ -227,6 +217,78 @@ def _maybe_postprocess_batch_instruction(
 
         return instruction
 
+    def _detect_many_measurement_scenario(self, instructions):
+        count = 0
+
+        for instruction in instructions:
+            if isinstance(instruction, Measurement):
+                count += 1
+                if count > 1:
+                    return True
+
+        return False
+
+    def _postprocess_many_measurement_sample(self, samples_on_modes):
+        unrolled_samples_on_modes = {}
+        for modes, subsample in samples_on_modes.items():
+            for mode, value in zip(modes, subsample):
+                unrolled_samples_on_modes[mode] = value
+
+        return [value for _, value in sorted(unrolled_samples_on_modes.items())]
+
+    def _execute_many_measurement_instructions(self, state, instructions, shots):
+        samples = []
+
+        def remap_modes(modes):
```

Comment:
These functions should be put into the class as staticmethods or otherwise organized outside of the function - they don't necessarily belong inside here.